src/collectors/proc.plugin/sys_class_power_supply.c

```diff
@@ -116,14 +130,83 @@ static void add_labels_to_power_supply(struct power_supply *ps, RRDSET *st) {
     rrdlabels_add(st->rrdlabels, "device", ps->name, RRDLABEL_SRC_AUTO);
 }
 
+static bool read_simple_property(struct simple_property *prop, bool keep_fds_open) {
+    char buffer[30 + 1];
+
+    if(unlikely(prop->fd == -1)) {
+        prop->fd = open(prop->filename, O_RDONLY | O_CLOEXEC, 0666);
+        if(unlikely(prop->fd == -1)) {
+            collector_error("Cannot open file '%s'", prop->filename);
+            return false;
+        }
+    }
+
+    if (likely(prop->fd != -1))
```

Comment:
I suggest instead to test `prop->fd != -1` again, you write it as an `else` for previous block.